### DISCUSSION TITLE
Way to freeze flame spread similar to freezing time for ramp input?

### DISCUSSION THREAD

**Discussion by bwklein** [firemodels/fds#13222] [Q&A]
I'm looking for a way to freeze the 'spread' of a fire growth that is based on the SPREAD_RATE parameter.
Similar to how we can freeze the time value input to a ramp for HRRPUA, is there a way to do this with the input to the spreading function.  Essentially, stopping the spread of the fire, but keeping the cells active that are within the spread radius.

**Comment by mcgratta** [comment]
I don't think there's a way to do this. Given a user-specified `SPREAD_RATE` FDS pre-computes the ignition time of the solid cells based on the assumption of a constant rate. What you want could be done with an additional parameter that limits the radius of the circle.

**Comment by mcgratta** [comment]
It would be easy to add `SPREAD_DISTANCE` to limit the extent of the fire spread. This could be handy for those who might want to create a circle of finite radius this is burning. I'll add and push up. Note the new parameter will be in `SURF` and `VENT` along with `SPREAD_RATE`.

**Comment by bwklein** [comment]
@mcgratta The use case that prompted this discussion was to have a control or device setpoint used to stop the spread from continuing.  I like the SPREAD_DISTANCE param, that will be a nice feature, and I can see many use cases for that. But it doesn't help for the intended use case to stop the spread during runtime.  Maybe there is something that can be added that takes a TRUE/FALSE input that can be linked to a DEVC_ID or CTRL_ID that can be used to stop the fire from spreading past the point of activation?

**Comment by drjfloyd** [comment]
Could use RAMP_V as way to have a variable spread rate since RAMP_Q
controls the fire. Then you could have control functions dictate the ramp
input or output.

**Reply by bwklein** [reply]
I like the `NO_UPDATE_CTRL_ID` concept that is used on devices that are inputs to ramp functions.  Something like this for a V_RAMP would achieve the desired use case.

**Comment by drjfloyd** [comment]
If we are going to mucking with SPREAD_RATE, a RAMP linkage would give
arbitrary ability to do whatever you wanted. This way we don't need to keep
adding keywords for each new way to do something with the rate.

**Reply by bwklein** [reply]
I will add that maybe a similar input param on the VENT input line would achieve the same effect.  NO_UPDATE_CTRL_ID and NO_UPDATE_DEVC_ID would be used to stop the spread and not require a RAMP_V since until that point the spread rate would be constant.  This would effectively set the SPREAD_RATE to 0m/s when TRUE, overriding the SPREAD_RATE value or whatever the RAMP_V value would be for that VENT at that time.

**Comment by mcgratta** [comment]
This is how `SPREAD_RATE` works now
```
   IF (VT%FIRE_SPREAD_RATE>0._EB) THEN
      DIST = SQRT((BC%X-VT%X0)**2 + (BC%Y-VT%Y0)**2 + (BC%Z-VT%Z0)**2)
      T_ACTIVATE = TT + DIST/VT%FIRE_SPREAD_RATE
   ENDIF
```
That is, each solid phase cell on the `VENT` is assigned an ignition time corresponding to the linear spread rate. I don't see how these controls or ramps would work.

**Comment by mcgratta** [comment]
I would add that I was just planning on adding `SPREAD_DISTANCE` which would be a limit on the expansion of the circle.

**Reply by drjfloyd** [reply]
With a RAMP multiplying the SPREAD_RATE you would just have the RAMP go to zero to stop spread.

The spread logic would need to move from init to wall.  Then for the VENT we would store the spread distance (the integtration of SPREAD_RATE*DT*RAMP value). Then for a given wall cell we would activate it if its DIST is greater then the spread distance in the same manner as if we had an IGNITION_TEMPERATURE set.  If the cell is already active we wouldn't both doing the check.

**Reply by mcgratta** [reply]
OK, let me ponder this. If I do it right, I can probably reduce the number of lines of code.
